prow/github/client.go

```diff
@@ -391,10 +392,36 @@ func (c *Client) BotName() (string, error) {
 			return "", fmt.Errorf("fetching bot name from GitHub: %v", err)
 		}
 		c.botName = u.Login
+		// email needs to be publicly accessible via the profile
+		// of the current account. Read below for more info
+		// https://developer.github.com/v3/users/#get-a-single-user
+		c.email = u.Email
 	}
 	return c.botName, nil
 }
 
+func (c *Client) Email() (string, error) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	if c.email == "" {
```

Comment:
can we shove this `if` block into a shared method? seems 100% the same between these two?

Reply:
sure, fixed.